src/test/java/org/jabref/logic/citationstyle/CitationStyleTest.java

```diff
@@ -37,8 +43,31 @@ void defaultCitation() {
     }
 
     @Test
-    void discoverCitationStylesNotNull() throws Exception {
+    void discoverCitationStylesNotNull() {
         List<CitationStyle> styleList = CitationStyle.discoverCitationStyles();
         assertNotNull(styleList);
     }
+
+    @ParameterizedTest
+    @MethodSource("citationStyleProvider")
```

Comment:
Typically, name the providing method the same as the testing method

```java
    @MethodSource
```

However, in your case, you are providing one source for multiple tests.

Please split up your test case in three:

Name the methods as follows:

- `citationStylePresent`
- `titleMatches`
- `numericPropertyMatches`

Do NOT use explanations for `assertEquals`. All of your text can be read out of the code.

Maybe, you are searching for the `name` parameter of `@ParamterizedTest` (https://stackoverflow.com/a/57894201/873282)?

Reply:
> Maybe, you are searching for the name parameter of @ParamterizedTest (https://stackoverflow.com/a/57894201/873282)?

This was an interesting read. Right now, we don't need it but can come in handy sometime.

Reply:
Updated: used same name for provider and test method uniformly throughout PR.